test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -189,24 +196,107 @@ public void testSegmentNotifications() throws Exception {
         assertEquals(2, segmentNotificationPostScale.getNumOfSegments());
     }
 
+    @Test(timeout = 40000)
+    public void testTargetRateNotifications() throws Exception {
```

Comment:
Isn't this test case similar to `io.pravega.test.integration.ReaderGroupNotificationTest#testSegmentNotifications` ? 
The only difference is that the new test waits for Stream to auto-scaled by the controller while the existing test manually increases the number of segments by invoking controller.scaleAPI(). 
Let me know if I missed anything 🤔

Reply:
Yes Sandeep, the difference is in the number of notifications. The former is fixed and the latter is based on rate. Thanks.